### PR TITLE
Added simple cookie classes. Fixed alot of other stuff. Working, can …

### PR DESCRIPTION
**Pull request by wside**
Begins work on #27

-Edited some DB format and added cookies table
-**User (or Company) Id is now stored in authentication table**

**made it so have to enter Id explicity when creating an authentication so need to get the Id using getUser or getCompany first from FE. e.g. say the Id you got back is 5, then do: 
curl -X POST http://localhost:8080/5/jcooldude/passwordhash/TRUE

-Made postings and applications auto-timestamp via mysql
-Made postings and applications return timestamp in GET methods
-Kept user timestamp generation in jdbc (for future salting purposes) and fixed it so it generates the correct time
-Made postings getRandom function for Luka to return 10 random postings

-Made CookieHandler and Cookie files
-getCookie checks against a provided cookie
-createCookie makes sure username and password exist in authentication table then generates a cookie from them (can discuss better hash function later) and sends back to front-end
-Auto purges: everyday, cookie table in DB destorys any cookie that is 2 days old or older

-updated http req doc

### CODE REVIEW

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/Authentication.java [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 @AutoMatter
 public interface Authentication {
+    Integer Id();
```
Comment (eigenfoo) [inline euphoria-service/src/main/java/edu/cooper/ece366/euphoria/Authentication.java]: Just to document for ourselves

```suggestion
    Integer Id();  // Either userId or companyId
```

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/CompanyHandlers.java [hunk 2]
```diff
@@ -28,7 +28,7 @@ public CompanyHandlers(final ObjectMapper objectMapper, final Config config) {
     public Stream<Route<AsyncHandler<Response<ByteString>>>> routes() {
         return Stream.of(
                 Route.sync("GET", "/company/<companyId>", this::getCompany),
-                Route.sync("POST", "/company/<name>/<website>/<description>",
+                Route.sync("POST", "/company/<name>/<website>/<description>/<username>/<passwordHash>/<isUser>",
```
Comment (eigenfoo) [inline euphoria-service/src/main/java/edu/cooper/ece366/euphoria/CompanyHandlers.java]: If we're posting to a company, we shouldn't need to have `isUser` as a path arg: it should be `False`.

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/UserHandlers.java [hunk 3]
```diff
@@ -29,7 +29,7 @@ public UserHandlers(final ObjectMapper objectMapper, final Config config) {
         return Stream.of(
                 Route.sync("GET", "/user/<userId>", this::getUser),
                 Route.sync("POST",
-                        "/user/<name>/<email>/<phoneNumber>/<educationLevel>/<description>",
+                        "/user/<name>/<email>/<phoneNumber>/<educationLevel>/<description>/<username>/<passwordHash>/<isUser>",
```
Comment (eigenfoo) [inline euphoria-service/src/main/java/edu/cooper/ece366/euphoria/UserHandlers.java]: Similar to above. Posting to the users URL should mean that `isUser` is `True`.

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/CookieHandlers.java [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    public List<Cookie> getCookie(final RequestContext rc) {
+        Cookie cookie = null;
+
+        try {
+            String cookieCheck = rc.pathArgs().get("cookieCheck");
+            Connection conn = DriverManager.getConnection(
+                    config.getString("mysql.jdbc"),
+                    config.getString("mysql.user"),
+                    config.getString("mysql.password"));
+            String sqlQuery = "SELECT * FROM coookies WHERE (cookie) IN ((?))";
```
Comment (eigenfoo) [inline euphoria-service/src/main/java/edu/cooper/ece366/euphoria/CookieHandlers.java]: Typo?

```suggestion
            String sqlQuery = "SELECT * FROM cookies WHERE (cookie) IN ((?))";
```

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/UserHandlers.java [hunk 5]
```diff
@@ -88,9 +88,22 @@ public UserHandlers(final ObjectMapper objectMapper, final Config config) {
             ps.setString(3, phoneNumber);
             ps.setString(4, educationLevel.toString());
             ps.setString(5, description);
-            Date date = new Date();
-            ps.setObject(6, date.toInstant().atZone(ZoneId.of("UTC")).toLocalDate());
-            ps.executeUpdate();
+            //did not set to auto-timestamp by mysql in case want to salt. Does put correct UMT timestamp now, though
```
Comment (eigenfoo) [inline euphoria-service/src/main/java/edu/cooper/ece366/euphoria/UserHandlers.java]: Is UMT and UTC the same thing? Google doesn't seem to know what UMT is...

**Review by eigenfoo — CHANGES_REQUESTED** [review]
~Did you check to make sure the cookies all work locally?~

~Also, can you edit the `http-requests.md` file with the new http requests for cookies?~

Just some docstrings and some issues on the urls for the http requests, looks great otherwise. Also some style issues, but those aren't that important and I can fix them in a subsequent PR.

- `Id` should be `id`
- ~`UpdateStatus` should be `updateStatus`~

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/UserHandlers.java (hunk 5 above)
Comment (wside) [reply euphoria-service/src/main/java/edu/cooper/ece366/euphoria/UserHandlers.java]: i meant to write UTC

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/CookieHandlers.java (hunk 4 above)
Comment (wside) [reply euphoria-service/src/main/java/edu/cooper/ece366/euphoria/CookieHandlers.java]: oops i forgot to test this my bad. its good now.

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/CompanyHandlers.java (hunk 2 above)
Comment (wside) [reply euphoria-service/src/main/java/edu/cooper/ece366/euphoria/CompanyHandlers.java]: changed so this is now a mute point

File: euphoria-service/src/main/java/edu/cooper/ece366/euphoria/UserHandlers.java (hunk 3 above)
Comment (wside) [reply euphoria-service/src/main/java/edu/cooper/ece366/euphoria/UserHandlers.java]: changed so this is now a mute point

**Review by eigenfoo — APPROVED** [review]

**Pull request merged**